### PR TITLE
feat: Flexible Error/Exception handling

### PR DESCRIPTION
**Pull request by vakrilov**
Implements #5914

### CODE REVIEW

File: tests/app/trace/trace-error-tests.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import { constants } from "zlib";
+import { trace } from "tns-core-modules/profiling/profiling";
+
+let cachedErrorHandler: ErrorHandler;
+export function setUpModule() {
+    cachedErrorHandler = getErrorHandler();
+}
+
+// before each
+export function tearDown() {
```
Comment (MartoYankov) [inline tests/app/trace/trace-error-tests.ts]: Technically, this is executed after each test. There is a setUp() function for before each.

File: tests/app/trace/trace-error-tests.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    TKUnit.assert(called, "trace.error() should call handler")
+}
+
+export function test_trace_error_should_create_error_from_string() {
+    let called = false;
+    setErrorHandler({
+        handlerError(error) {
+            called = true;
+            TKUnit.assert(error instanceof Error, "trace.error() wrap string in error")
```
Comment (MartoYankov) [inline tests/app/trace/trace-error-tests.ts]: Even if it's fine in this case, I think it's dangerous to use asserts in callbacks. It's better to cache the value and assert it after the assertion of the `called` variable. In cases where the callbacks are async asserting in them breaks the tests.

File: tns-core-modules/trace/trace.d.ts [hunk 3]
```diff
@@ -63,6 +63,11 @@ export function isCategorySet(category: string): boolean;
  */
 export function write(message: any, category: string, type?: number);
 
+/**
+ * Passes an error ot the registered ErrorHandler
```
Comment (MartoYankov) [inline tns-core-modules/trace/trace.d.ts]: typo - "ot" -> "to"

File: tests/app/trace/trace-error-tests.ts (hunk 1 above)
Comment (vakrilov) [reply tests/app/trace/trace-error-tests.ts]: yup - I want to cache the original `ErrorHandler` once and clean-up after each test to make sure the test does not leak side effects

File: HandlingErrors.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
+# Handling Errors in NativeScript Core Modules
+
+One big difference between web-app and NativeScript app is that throwing and `Error` in NativeScript app cause the app to **crash**. Such crashes can be a serious problem for a production applications as they the application credibility and drive away customers.
```
Comment (vchimev) [inline HandlingErrors.md]: Such crashes can be a serious problem for ~~a~~ production applications as they **hurt** the application credibility ...

**Pull request merged**